src/roman_datamodels/stnode/_schema.py

```diff
@@ -187,7 +191,32 @@ class SchemaType(enum.Enum):
 _NUMERIC_KEYWORDS = {"multipleOf", "maximum", "exclusiveMaximum", "minimum"}
 
 
-class Builder:
+@runtime_checkable
+class Build(Protocol):
+    """
+    Protocol for implementing building of a node from a schema.
+    """
+
+    @abstractmethod
+    def build(self, schema: Mapping[str, Any], defaults: Any | None = None) -> Mapping[str, Any] | Sequence[Any]:
+        """
+        Build the data for a node from a schema and optional defaults.
+
+        Parameters
+        ----------
+        schema : Mapping[str, Any]
+            The schema to build from.
+        defaults : Any | None
+            The defaults to use when building.
+
+        Returns
+        -------
+        Mapping[str, Any] or Sequence[Any]
+            The built data for a node based on the schema using the defaults where possible.
+        """
+
+
+class Builder(Build):
```

Comment:
Not that inheritance from a `Protocol` is the same as inheriting from `abc.Abstract`; however, one does not have to inherit from it if one does not have to. The `@runtime_checkable` just allows `isinstance` to check if an instance of an object satisfies the protocol. In fact all the `collections.abc` are actually just `Protocol` objects with default implementations for many of the methods.